infra/lib/ci-cluster.ts

```diff
@@ -34,36 +34,58 @@ export class CICluster extends Construct {
 
     const clusterVersion = eks.KubernetesVersion.V1_23;
 
-    const mngProps: blueprints.MngClusterProviderProps = {
-      minSize: 2,
-      maxSize: 8,
-      desiredSize: 2,
-      diskSize: 150,
-      version: clusterVersion,
-      instanceTypes: [
-        ec2.InstanceType.of(ec2.InstanceClass.M5, ec2.InstanceSize.XLARGE8),
+    const karpenterAddonProps = {
+      requirements: [
+          {
+            key: 'node.kubernetes.io/instance-type',
+            op: "In" as const,
+            vals: ['m5.xlarge','m5.2xlarge','m5.4xlarge','m5.8xlarge'],
+          },
+          {
+            key: 'kubernetes.io/arch',
+            op: "In" as const,
+            vals: ['amd64']
+          },
+          {
+            key: 'karpenter.sh/capacity-type',
+            op: "In" as const,
+            vals: ['on-demand']
+          },
       ],
-      amiType: eks.NodegroupAmiType.AL2_X86_64,
-      nodeGroupCapacityType: eks.CapacityType.ON_DEMAND,
-    };
+      subnetTags: {
+        "Name": "TestCIStack/CIClusterConstruct/TestInfraCluster/TestInfraCluster-vpc/PrivateSubnet*",
+      },
+      securityGroupTags: {
+        "kubernetes.io/cluster/TestInfraCluster": "owned",
+      },
+      taints: [{
+        key: "workload",
+        value: "test",
+        effect: "NoSchedule" as const,
```

Comment:
I don't see matching tolerations on our `ProwJob` pods, are we sure we'd able to schedule pods on these nodes?

Reply:
Removed the taint; we can revisit using taints + tolerations in a later iteration